Lib/asyncore.py

```diff
@@ -144,20 +145,20 @@ def poll(timeout=0.0, map=None):
         r, w, e = select.select(r, w, e, timeout)
 
         for fd in r:
-            obj = map.get(fd)
-            if obj is None:
+            obj = map_copy.get(fd)
+            if fd not in map or map.get(fd) != obj:
```

Comment:
Same, the map cannot change while the single thread that may access the map is block on select().

And, getting the fd from the map is good check in case the fd was removed from the map by a dispatcher.